vm-validator/src/vm_validator.rs

```diff
@@ -21,36 +22,61 @@ use storage_interface::{state_view::VerifiedStateView, DbReader};
 #[path = "unit_tests/vm_validator_test.rs"]
 mod vm_validator_test;
 
-pub trait TransactionValidation: Send + Sync + Clone {
+pub trait TransactionValidation: Send + Sync {
     type ValidationInstance: diem_vm::VMValidator;
 
     /// Validate a txn from client
     fn validate_transaction(&self, _txn: SignedTransaction) -> Result<VMValidatorResult>;
 
     /// Restart the transaction validation instance
     fn restart(&mut self, config: OnChainConfigPayload) -> Result<()>;
+
+    /// Notify about new commit
+    fn notify_commit(&mut self);
+}
+
+#[ouroboros::self_referencing(pub_extras)]
```

Comment:
not really necessary after https://github.com/diem/diem/pull/9552/commits/7ca5a9056aea740667c2a681158fe14903f60b94 

will try to land that.